CI: bpf: TestBatchIterator fails: expect iterate to return key=12

## CI failure

https://github.com/cilium/cilium/actions/runs/12795392193/job/35674240992#step:17:14341
Hit on PR https://github.com/cilium/cilium/pull/36602, but this looks unrelated.

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/pkg/bpf   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestBatchIterator (24.71s)
--- FAIL: TestBatchIterator/size=1024_mapSize=1024#01 (0.01s)

    map_linux_test.go:983: 
        	Error Trace:	/host/pkg/bpf/map_linux_test.go:983
        	            				/host/pkg/bpf/map_linux_test.go:1028
        	Error:      	Not equal: 
        	            	expected: 1024
        	            	actual  : 896
        	Test:       	TestBatchIterator/size=1024_mapSize=1024#01
    map_linux_test.go:986: 
        	Error Trace:	/host/pkg/bpf/map_linux_test.go:986
        	            				/host/pkg/bpf/map_linux_test.go:1028
        	Error:      	sets.Set[int]{0:sets.Empty{}, 1:sets.Empty{}, 2:sets.Empty{}, 3:sets.Empty{}, 4:sets.Empty{}, 5:sets.Empty{}, 6:sets.Empty{}, 7:sets.Empty{}, 8:sets.Empty{}, 9:sets.Empty{}, 10:sets.Empty{}, 11:sets.Empty{}, 140:sets.Empty{}, 141:sets.Empty{}, 142:sets.Empty{}, 143:sets.Empty{}, 144:sets.Empty{}, 145:sets.Empty{}, 146:sets.Empty{}, 147:sets.Empty{}, 148:sets.Empty{}, 149:sets.Empty{}, 150:sets.Empty{}, 151:sets.Empty{}, 152:sets.Empty{}, 153:sets.Empty{}, 154:sets.Empty{}, 155:sets.Empty{}, 156:sets.Empty{}, 157:sets.Empty{}, 158:sets.Empty{}, 159:sets.Empty{}, 160:sets.Empty{}, 161:sets.Empty{}, 162:sets.Empty{}, 163:sets.Empty{}, 164:sets.Empty{}, 165:sets.Empty{}, 166:sets.Empty{}, 167:sets.Empty{}, 168:sets.Empty{}, 169:sets.Empty{}, 170:sets.Empty{}, 171:sets.Empty{}, 172:sets.Empty{}, 173:sets.Empty{}, 174:sets.Empty{}, 175:sets.Empty{}, 176:sets.Empty{}, 177:sets.Empty{}, 178:sets.Empty{}, 179:sets.Empty{}, 180:sets.Empty{}, 181:sets.Empty{}, 182:sets.Empty{}, 183:sets.Empty{}, 184:sets.Empty{}, 185:sets.Empty{}, 186:sets.Empty{}, 187:sets.Empty{}, 188:sets.Empty{}, 189:sets.Empty{}, 190:sets.Empty{}, 191:sets.Empty{}, 192:sets.Empty{}, 193:sets.Empty{}, 194:sets.Empty{}, 195:sets.Empty{}, 196:sets.Empty{}, 197:sets.Empty{}, 198:sets.Empty{}, 199:sets.Empty{}, 200:sets.Empty{}, 201:sets.Empty{}, 202:sets.Empty{}, 203:sets.Empty{}, 204:sets.Empty{}, 205:sets.Empty{}, 206:sets.Empty{}, 207:sets.Empty{}, 208:sets.Empty{}, 209:sets.Empty{}, 210:sets.Empty{}, 211:sets.Empty{}, 212:sets.Empty{}, 213:sets.Empty{}, 214:sets.Empty{}, 215:sets.Empty{}, 216:sets.Empty{}, 217:sets.Empty{}, 218:sets.Empty{}, 219:sets.Empty{}, 220:sets.Empty{}, 221:sets.Empty{}, 222:sets.Empty{}, 223:sets.Empty{}, 224:sets.Empty{}, 225:sets.Empty{}, 226:sets.Empty{}, 227:sets.Empty{}, 228:sets.Empty{}, 229:sets.Empty{}, 230:sets.Empty{}, 231:sets.Empty{}, 232:sets.Empty{}, 233:sets.Empty{}, 234:sets.Empty{}, 235:sets.Empty{}, 236:sets.Empty{}, 237:sets.Empty{}, 238:sets.Empty{}, 239:sets.Empty{}, 240:sets.Empty{}, 241:sets.Empty{}, 242:sets.Empty{}, 243:sets.Empty{}, 244:sets.Empty{}, 245:sets.Empty{}, 246:sets.Empty{}, 247:sets.Empty{}, 248:sets.Empty{}, 249:sets.Empty{}, 250:sets.Empty{}, 251:sets.Empty{}, 252:sets.Empty{}, 253:sets.Empty{}, 254:sets.Empty{}, 255:sets.Empty{}, 256:sets.Empty{}, 257:sets.Empty{}, 258:sets.Empty{}, 259:sets.Empty{}, 260:sets.Empty{}, 261:sets.Empty{}, 262:sets.Empty{}, 263:sets.Empty{}, 264:sets.Empty{}, 265:sets.Empty{}, 266:sets.Empty{}, 267:sets.Empty{}, 268:sets.Empty{}, 269:sets.Empty{}, 270:sets.Empty{}, 271:sets.Empty{}, 272:sets.Empty{}, 273:sets.Empty{}, 274:sets.Empty{}, 275:sets.Empty{}, 276:sets.Empty{}, 277:sets.Empty{}, 278:sets.Empty{}, 279:sets.Empty{}, 280:sets.Empty{}, 281:sets.Empty{}, 282:sets.Empty{}, 283:sets.Empty{}, 284:sets.Empty{}, 285:sets.Empty{}, 286:sets.Empty{}, 287:sets.Empty{}, 288:sets.Empty{}, 289:sets.Empty{}, 290:sets.Empty{}, 291:sets.Empty{}, 292:sets.Empty{}, 293:sets.Empty{}, 294:sets.Empty{}, 295:sets.Empty{}, 296:sets.Empty{}, 297:sets.Empty{}, 298:sets.Empty{}, 299:sets.Empty{}, 300:sets.Empty{}, 301:sets.Empty{}, 302:sets.Empty{}, 303:sets.Empty{}, 304:sets.Empty{}, 305:sets.Empty{}, 306:sets.Empty{}, 307:sets.Empty{}, 308:sets.Empty{}, 309:sets.Empty{}, 310:sets.Empty{}, 311:sets.Empty{}, 312:sets.Empty{}, 313:sets.Empty{}, 314:sets.Empty{}, 315:sets.Empty{}, 316:sets.Empty{}, 317:sets.Empty{}, 318:sets.Empty{}, 319:sets.Empty{}, 320:sets.Empty{}, 321:sets.Empty{}, 322:sets.Empty{}, 323:sets.Empty{}, 324:sets.Empty{}, 325:sets.Empty{}, 326:sets.Empty{}, 327:sets.Empty{}, 328:sets.Empty{}, 329:sets.Empty{}, 330:sets.Empty{}, 331:sets.Empty{}, 332:sets.Empty{}, 333:sets.Empty{}, 334:sets.Empty{}, 335:sets.Empty{}, 336:sets.Empty{}, 337:sets.Empty{}, 338:sets.Empty{}, 339:sets.Empty{}, 340:sets.Empty{}, 341:sets.Empty{}, 342:sets.Empty{}, 343:sets.Empty{}, 344:sets.Empty{}, 345:sets.Empty{}, 346:sets.Empty{}, 347:sets.Empty{}, 348:sets.Empty{}, 349:sets.Empty{}, 350:sets.Empty{}, 351:sets.Empty{}, 352:sets.Empty{}, 353:sets.Empty{}, 354:sets.Empty{}, 355:sets.Empty{}, 356:sets.Empty{}, 357:sets.Empty{}, 358:sets.Empty{}, 359:sets.Empty{}, 360:sets.Empty{}, 361:sets.Empty{}, 362:sets.Empty{}, 363:sets.Empty{}, 364:sets.Empty{}, 365:sets.Empty{}, 366:sets.Empty{}, 367:sets.Empty{}, 368:sets.Empty{}, 369:sets.Empty{}, 370:sets.Empty{}, 371:sets.Empty{}, 372:sets.Empty{}, 373:sets.Empty{}, 374:sets.Empty{}, 375:sets.Empty{}, 376:sets.Empty{}, 377:sets.Empty{}, 378:sets.Empty{}, 379:sets.Empty{}, 380:sets.Empty{}, 381:sets.Empty{}, 382:sets.Empty{}, 383:sets.Empty{}, 384:sets.Empty{}, 385:sets.Empty{}, 386:sets.Empty{}, 387:sets.Empty{}, 388:sets.Empty{}, 389:sets.Empty{}, 390:sets.Empty{}, 391:sets.Empty{}, 392:sets.Empty{}, 393:sets.Empty{}, 394:sets.Empty{}, 395:sets.Empty{}, 396:sets.Empty{}, 397:sets.Empty{}, 398:sets.Empty{}, 399:sets.Empty{}, 400:sets.Empty{}, 401:sets.Empty{}, 402:sets.Empty{}, 403:sets.Empty{}, 404:sets.Empty{}, 405:sets.Empty{}, 406:sets.Empty{}, 407:sets.Empty{}, 408:sets.Empty{}, 409:sets.Empty{}, 410:sets.Empty{}, 411:sets.Empty{}, 412:sets.Empty{}, 413:sets.Empty{}, 414:sets.Empty{}, 415:sets.Empty{}, 416:sets.Empty{}, 417:sets.Empty{}, 418:sets.Empty{}, 419:sets.Empty{}, 420:sets.Empty{}, 421:sets.Empty{}, 422:sets.Empty{}, 423:sets.Empty{}, 424:sets.Empty{}, 425:sets.Empty{}, 426:sets.Empty{}, 427:sets.Empty{}, 428:sets.Empty{}, 429:sets.Empty{}, 430:sets.Empty{}, 431:sets.Empty{}, 432:sets.Empty{}, 433:sets.Empty{}, 434:sets.Empty{}, 435:sets.Empty{}, 436:sets.Empty{}, 437:sets.Empty{}, 438:sets.Empty{}, 439:sets.Empty{}, 440:sets.Empty{}, 441:sets.Empty{}, 442:sets.Empty{}, 443:sets.Empty{}, 444:sets.Empty{}, 445:sets.Empty{}, 446:sets.Empty{}, 447:sets.Empty{}, 448:sets.Empty{}, 449:sets.Empty{}, 450:sets.Empty{}, 451:sets.Empty{}, 452:sets.Empty{}, 453:sets.Empty{}, 454:sets.Empty{}, 455:sets.Empty{}, 456:sets.Empty{}, 457:sets.Empty{}, 458:sets.Empty{}, 459:sets.Empty{}, 460:sets.Empty{}, 461:sets.Empty{}, 462:sets.Empty{}, 463:sets.Empty{}, 464:sets.Empty{}, 465:sets.Empty{}, 466:sets.Empty{}, 467:sets.Empty{}, 468:sets.Empty{}, 469:sets.Empty{}, 470:sets.Empty{}, 471:sets.Empty{}, 472:sets.Empty{}, 473:sets.Empty{}, 474:sets.Empty{}, 475:sets.Empty{}, 476:sets.Empty{}, 477:sets.Empty{}, 478:sets.Empty{}, 479:sets.Empty{}, 480:sets.Empty{}, 481:sets.Empty{}, 482:sets.Empty{}, 483:sets.Empty{}, 484:sets.Empty{}, 485:sets.Empty{}, 486:sets.Empty{}, 487:sets.Empty{}, 488:sets.Empty{}, 489:sets.Empty{}, 490:sets.Empty{}, 491:sets.Empty{}, 492:sets.Empty{}, 493:sets.Empty{}, 494:sets.Empty{}, 495:sets.Empty{}, 496:sets.Empty{}, 497:sets.Empty{}, 498:sets.Empty{}, 499:sets.Empty{}, 500:sets.Empty{}, 501:sets.Empty{}, 502:sets.Empty{}, 503:sets.Empty{}, 504:sets.Empty{}, 505:sets.Empty{}, 506:sets.Empty{}, 507:sets.Empty{}, 508:sets.Empty{}, 509:sets.Empty{}, 510:sets.Empty{}, 511:sets.Empty{}, 512:sets.Empty{}, 513:sets.Empty{}, 514:sets.Empty{}, 515:sets.Empty{}, 516:sets.Empty{}, 517:sets.Empty{}, 518:sets.Empty{}, 519:sets.Empty{}, 520:sets.Empty{}, 521:sets.Empty{}, 522:sets.Empty{}, 523:sets.Empty{}, 524:sets.Empty{}, 525:sets.Empty{}, 526:sets.Empty{}, 527:sets.Empty{}, 528:sets.Empty{}, 529:sets.Empty{}, 530:sets.Empty{}, 531:sets.Empty{}, 532:sets.Empty{}, 533:sets.Empty{}, 534:sets.Empty{}, 535:sets.Empty{}, 536:sets.Empty{}, 537:sets.Empty{}, 538:sets.Empty{}, 539:sets.Empty{}, 540:sets.Empty{}, 541:sets.Empty{}, 542:sets.Empty{}, 543:sets.Empty{}, 544:sets.Empty{}, 545:sets.Empty{}, 546:sets.Empty{}, 547:sets.Empty{}, 548:sets.Empty{}, 549:sets.Empty{}, 550:sets.Empty{}, 551:sets.Empty{}, 552:sets.Empty{}, 553:sets.Empty{}, 554:sets.Empty{}, 555:sets.Empty{}, 556:sets.Empty{}, 557:sets.Empty{}, 558:sets.Empty{}, 559:sets.Empty{}, 560:sets.Empty{}, 561:sets.Empty{}, 562:sets.Empty{}, 563:sets.Empty{}, 564:sets.Empty{}, 565:sets.Empty{}, 566:sets.Empty{}, 567:sets.Empty{}, 568:sets.Empty{}, 569:sets.Empty{}, 570:sets.Empty{}, 571:sets.Empty{}, 572:sets.Empty{}, 573:sets.Empty{}, 574:sets.Empty{}, 575:sets.Empty{}, 576:sets.Empty{}, 577:sets.Empty{}, 578:sets.Empty{}, 579:sets.Empty{}, 580:sets.Empty{}, 581:sets.Empty{}, 582:sets.Empty{}, 583:sets.Empty{}, 584:sets.Empty{}, 585:sets.Empty{}, 586:sets.Empty{}, 587:sets.Empty{}, 588:sets.Empty{}, 589:sets.Empty{}, 590:sets.Empty{}, 591:sets.Empty{}, 592:sets.Empty{}, 593:sets.Empty{}, 594:sets.Empty{}, 595:sets.Empty{}, 596:sets.Empty{}, 597:sets.Empty{}, 598:sets.Empty{}, 599:sets.Empty{}, 600:sets.Empty{}, 601:sets.Empty{}, 602:sets.Empty{}, 603:sets.Empty{}, 604:sets.Empty{}, 605:sets.Empty{}, 606:sets.Empty{}, 607:sets.Empty{}, 608:sets.Empty{}, 609:sets.Empty{}, 610:sets.Empty{}, 611:sets.Empty{}, 612:sets.Empty{}, 613:sets.Empty{}, 614:sets.Empty{}, 615:sets.Empty{}, 616:sets.Empty{}, 617:sets.Empty{}, 618:sets.Empty{}, 619:sets.Empty{}, 620:sets.Empty{}, 621:sets.Empty{}, 622:sets.Empty{}, 623:sets.Empty{}, 624:sets.Empty{}, 625:sets.Empty{}, 626:sets.Empty{}, 627:sets.Empty{}, 628:sets.Empty{}, 629:sets.Empty{}, 630:sets.Empty{}, 631:sets.Empty{}, 632:sets.Empty{}, 633:sets.Empty{}, 634:sets.Empty{}, 635:sets.Empty{}, 636:sets.Empty{}, 637:sets.Empty{}, 638:sets.Empty{}, 639:sets.Empty{}, 640:sets.Empty{}, 641:sets.Empty{}, 642:sets.Empty{}, 643:sets.Empty{}, 644:sets.Empty{}, 645:sets.Empty{}, 646:sets.Empty{}, 647:sets.Empty{}, 648:sets.Empty{}, 649:sets.Empty{}, 650:sets.Empty{}, 651:sets.Empty{}, 652:sets.Empty{}, 653:sets.Empty{}, 654:sets.Empty{}, 655:sets.Empty{}, 656:sets.Empty{}, 657:sets.Empty{}, 658:sets.Empty{}, 659:sets.Empty{}, 660:sets.Empty{}, 661:sets.Empty{}, 662:sets.Empty{}, 663:sets.Empty{}, 664:sets.Empty{}, 665:sets.Empty{}, 666:sets.Empty{}, 667:sets.Empty{}, 668:sets.Empty{}, 669:sets.Empty{}, 670:sets.Empty{}, 671:sets.Empty{}, 672:sets.Empty{}, 673:sets.Empty{}, 674:sets.Empty{}, 675:sets.Empty{}, 676:sets.Empty{}, 677:sets.Empty{}, 678:sets.Empty{}, 679:sets.Empty{}, 680:sets.Empty{}, 681:sets.Empty{}, 682:sets.Empty{}, 683:sets.Empty{}, 684:sets.Empty{}, 685:sets.Empty{}, 686:sets.Empty{}, 687:sets.Empty{}, 688:sets.Empty{}, 689:sets.Empty{}, 690:sets.Empty{}, 691:sets.Empty{}, 692:sets.Empty{}, 693:sets.Empty{}, 694:sets.Empty{}, 695:sets.Empty{}, 696:sets.Empty{}, 697:sets.Empty{}, 698:sets.Empty{}, 699:sets.Empty{}, 700:sets.Empty{}, 701:sets.Empty{}, 702:sets.Empty{}, 703:sets.Empty{}, 704:sets.Empty{}, 705:sets.Empty{}, 706:sets.Empty{}, 707:sets.Empty{}, 708:sets.Empty{}, 709:sets.Empty{}, 710:sets.Empty{}, 711:sets.Empty{}, 712:sets.Empty{}, 713:sets.Empty{}, 714:sets.Empty{}, 715:sets.Empty{}, 716:sets.Empty{}, 717:sets.Empty{}, 718:sets.Empty{}, 719:sets.Empty{}, 720:sets.Empty{}, 721:sets.Empty{}, 722:sets.Empty{}, 723:sets.Empty{}, 724:sets.Empty{}, 725:sets.Empty{}, 726:sets.Empty{}, 727:sets.Empty{}, 728:sets.Empty{}, 729:sets.Empty{}, 730:sets.Empty{}, 731:sets.Empty{}, 732:sets.Empty{}, 733:sets.Empty{}, 734:sets.Empty{}, 735:sets.Empty{}, 736:sets.Empty{}, 737:sets.Empty{}, 738:sets.Empty{}, 739:sets.Empty{}, 740:sets.Empty{}, 741:sets.Empty{}, 742:sets.Empty{}, 743:sets.Empty{}, 744:sets.Empty{}, 745:sets.Empty{}, 746:sets.Empty{}, 747:sets.Empty{}, 748:sets.Empty{}, 749:sets.Empty{}, 750:sets.Empty{}, 751:sets.Empty{}, 752:sets.Empty{}, 753:sets.Empty{}, 754:sets.Empty{}, 755:sets.Empty{}, 756:sets.Empty{}, 757:sets.Empty{}, 758:sets.Empty{}, 759:sets.Empty{}, 760:sets.Empty{}, 761:sets.Empty{}, 762:sets.Empty{}, 763:sets.Empty{}, 764:sets.Empty{}, 765:sets.Empty{}, 766:sets.Empty{}, 767:sets.Empty{}, 768:sets.Empty{}, 769:sets.Empty{}, 770:sets.Empty{}, 771:sets.Empty{}, 772:sets.Empty{}, 773:sets.Empty{}, 774:sets.Empty{}, 775:sets.Empty{}, 776:sets.Empty{}, 777:sets.Empty{}, 778:sets.Empty{}, 779:sets.Empty{}, 780:sets.Empty{}, 781:sets.Empty{}, 782:sets.Empty{}, 783:sets.Empty{}, 784:sets.Empty{}, 785:sets.Empty{}, 786:sets.Empty{}, 787:sets.Empty{}, 788:sets.Empty{}, 789:sets.Empty{}, 790:sets.Empty{}, 791:sets.Empty{}, 792:sets.Empty{}, 793:sets.Empty{}, 794:sets.Empty{}, 795:sets.Empty{}, 796:sets.Empty{}, 797:sets.Empty{}, 798:sets.Empty{}, 799:sets.Empty{}, 800:sets.Empty{}, 801:sets.Empty{}, 802:sets.Empty{}, 803:sets.Empty{}, 804:sets.Empty{}, 805:sets.Empty{}, 806:sets.Empty{}, 807:sets.Empty{}, 808:sets.Empty{}, 809:sets.Empty{}, 810:sets.Empty{}, 811:sets.Empty{}, 812:sets.Empty{}, 813:sets.Empty{}, 814:sets.Empty{}, 815:sets.Empty{}, 816:sets.Empty{}, 817:sets.Empty{}, 818:sets.Empty{}, 819:sets.Empty{}, 820:sets.Empty{}, 821:sets.Empty{}, 822:sets.Empty{}, 823:sets.Empty{}, 824:sets.Empty{}, 825:sets.Empty{}, 826:sets.Empty{}, 827:sets.Empty{}, 828:sets.Empty{}, 829:sets.Empty{}, 830:sets.Empty{}, 831:sets.Empty{}, 832:sets.Empty{}, 833:sets.Empty{}, 834:sets.Empty{}, 835:sets.Empty{}, 836:sets.Empty{}, 837:sets.Empty{}, 838:sets.Empty{}, 839:sets.Empty{}, 840:sets.Empty{}, 841:sets.Empty{}, 842:sets.Empty{}, 843:sets.Empty{}, 844:sets.Empty{}, 845:sets.Empty{}, 846:sets.Empty{}, 847:sets.Empty{}, 848:sets.Empty{}, 849:sets.Empty{}, 850:sets.Empty{}, 851:sets.Empty{}, 852:sets.Empty{}, 853:sets.Empty{}, 854:sets.Empty{}, 855:sets.Empty{}, 856:sets.Empty{}, 857:sets.Empty{}, 858:sets.Empty{}, 859:sets.Empty{}, 860:sets.Empty{}, 861:sets.Empty{}, 862:sets.Empty{}, 863:sets.Empty{}, 864:sets.Empty{}, 865:sets.Empty{}, 866:sets.Empty{}, 867:sets.Empty{}, 868:sets.Empty{}, 869:sets.Empty{}, 870:sets.Empty{}, 871:sets.Empty{}, 872:sets.Empty{}, 873:sets.Empty{}, 874:sets.Empty{}, 875:sets.Empty{}, 876:sets.Empty{}, 877:sets.Empty{}, 878:sets.Empty{}, 879:sets.Empty{}, 880:sets.Empty{}, 881:sets.Empty{}, 882:sets.Empty{}, 883:sets.Empty{}, 884:sets.Empty{}, 885:sets.Empty{}, 886:sets.Empty{}, 887:sets.Empty{}, 888:sets.Empty{}, 889:sets.Empty{}, 890:sets.Empty{}, 891:sets.Empty{}, 892:sets.Empty{}, 893:sets.Empty{}, 894:sets.Empty{}, 895:sets.Empty{}, 896:sets.Empty{}, 897:sets.Empty{}, 898:sets.Empty{}, 899:sets.Empty{}, 900:sets.Empty{}, 901:sets.Empty{}, 902:sets.Empty{}, 903:sets.Empty{}, 904:sets.Empty{}, 905:sets.Empty{}, 906:sets.Empty{}, 907:sets.Empty{}, 908:sets.Empty{}, 909:sets.Empty{}, 910:sets.Empty{}, 911:sets.Empty{}, 912:sets.Empty{}, 913:sets.Empty{}, 914:sets.Empty{}, 915:sets.Empty{}, 916:sets.Empty{}, 917:sets.Empty{}, 918:sets.Empty{}, 919:sets.Empty{}, 920:sets.Empty{}, 921:sets.Empty{}, 922:sets.Empty{}, 923:sets.Empty{}, 924:sets.Empty{}, 925:sets.Empty{}, 926:sets.Empty{}, 927:sets.Empty{}, 928:sets.Empty{}, 929:sets.Empty{}, 930:sets.Empty{}, 931:sets.Empty{}, 932:sets.Empty{}, 933:sets.Empty{}, 934:sets.Empty{}, 935:sets.Empty{}, 936:sets.Empty{}, 937:sets.Empty{}, 938:sets.Empty{}, 939:sets.Empty{}, 940:sets.Empty{}, 941:sets.Empty{}, 942:sets.Empty{}, 943:sets.Empty{}, 944:sets.Empty{}, 945:sets.Empty{}, 946:sets.Empty{}, 947:sets.Empty{}, 948:sets.Empty{}, 949:sets.Empty{}, 950:sets.Empty{}, 951:sets.Empty{}, 952:sets.Empty{}, 953:sets.Empty{}, 954:sets.Empty{}, 955:sets.Empty{}, 956:sets.Empty{}, 957:sets.Empty{}, 958:sets.Empty{}, 959:sets.Empty{}, 960:sets.Empty{}, 961:sets.Empty{}, 962:sets.Empty{}, 963:sets.Empty{}, 964:sets.Empty{}, 965:sets.Empty{}, 966:sets.Empty{}, 967:sets.Empty{}, 968:sets.Empty{}, 969:sets.Empty{}, 970:sets.Empty{}, 971:sets.Empty{}, 972:sets.Empty{}, 973:sets.Empty{}, 974:sets.Empty{}, 975:sets.Empty{}, 976:sets.Empty{}, 977:sets.Empty{}, 978:sets.Empty{}, 979:sets.Empty{}, 980:sets.Empty{}, 981:sets.Empty{}, 982:sets.Empty{}, 983:sets.Empty{}, 984:sets.Empty{}, 985:sets.Empty{}, 986:sets.Empty{}, 987:sets.Empty{}, 988:sets.Empty{}, 989:sets.Empty{}, 990:sets.Empty{}, 991:sets.Empty{}, 992:sets.Empty{}, 993:sets.Empty{}, 994:sets.Empty{}, 995:sets.Empty{}, 996:sets.Empty{}, 997:sets.Empty{}, 998:sets.Empty{}, 999:sets.Empty{}, 1000:sets.Empty{}, 1001:sets.Empty{}, 1002:sets.Empty{}, 1003:sets.Empty{}, 1004:sets.Empty{}, 1005:sets.Empty{}, 1006:sets.Empty{}, 1007:sets.Empty{}, 1008:sets.Empty{}, 1009:sets.Empty{}, 1010:sets.Empty{}, 1011:sets.Empty{}, 1012:sets.Empty{}, 1013:sets.Empty{}, 1014:sets.Empty{}, 1015:sets.Empty{}, 1016:sets.Empty{}, 1017:sets.Empty{}, 1018:sets.Empty{}, 1019:sets.Empty{}, 1020:sets.Empty{}, 1021:sets.Empty{}, 1022:sets.Empty{}, 1023:sets.Empty{}} does not contain 12
        	Test:       	TestBatchIterator/size=1024_mapSize=1024#01
        	Messages:   	expect iterate to return key=12
```

--- a/pkg/bpf/map_linux.go
+++ b/pkg/bpf/map_linux.go
@@ -957,7 +957,7 @@ func (bi *BatchIterator[KT, VT, KP, VP]) IterateAll(ctx context.Context, opts ..
 				// If we receive ENOSPC failures, we will try to recover by growing the batch buffer
 				// size (up to some max number of retries - default: 3) and retrying the iteration.
 				//
-				// [1] https://elixir.bootlin.com/linux/latest/source/kernel/bpf/hashtab.c#L1776
+				// [1] https://elixir.bootlin.com/linux/v6.12.6/source/kernel/bpf/hashtab.c#L1807-L1809
 				//
 				// Note: If this failure happens during the bpf syscall, it is expected that the underlying
 				// cursor will not have been swapped - meaning that we can retry the iteration at the same cursor.

--- a/pkg/bpf/map_linux_test.go
+++ b/pkg/bpf/map_linux_test.go
@@ -995,9 +995,16 @@ func TestBatchIterator(t *testing.T) {
 		mapSize int
 		size    int
 		opts    []BatchIteratorOpt[TestKey, TestValue, *TestKey, *TestValue]
+		// LRU hash maps aren't totally safe to test like this, even if you're
+		// within the max map size number of elements, in practice the kernel
+		// will occasionally do a LRU eviction causing failures.
+		// Setting the max size appears to make this safe enough (test up to a
+		// test million runs) that we can run a subset of tests on LRU.
+		unsafeLRU bool
 	}{
-		{10, 10, nil},
-		{1024, 1024, nil},
+		{10, 10, nil, true},
+		{1024, 1024, nil, true},
+		{1048576, 1024, nil, false}, // Max map size much larger means no chance of LRU eviction.
 		// Setup iteration that starts with batch size of 1, this is bound to fail at some point
 		// so this will test if the chunk size growth retry loop works correctly.
 		{
@@ -1005,27 +1012,27 @@ func TestBatchIterator(t *testing.T) {
 			mapSize: 1 << 13,
 			opts: []BatchIteratorOpt[TestKey, TestValue, *TestKey, *TestValue]{
 				WithMaxRetries[TestKey, TestValue](13), WithStartingChunkSize[TestKey, TestValue](1)},
+			unsafeLRU: true,
 		},
 		{
 			size:    1,
 			mapSize: 1 << 12,
 		},
 		{
-			size:    1 << 8,
-			mapSize: 1 << 8,
-		},
-		{
 			size:    1 << 12,
 			mapSize: 1 << 12,
 			opts: []BatchIteratorOpt[TestKey, TestValue, *TestKey, *TestValue]{
 				WithMaxRetries[TestKey, TestValue](1), WithStartingChunkSize[TestKey, TestValue](1 << 13)},
+			unsafeLRU: true,
 		},
 	} {
 		t.Run(fmt.Sprintf("size=%d mapSize=%d", test.size, test.mapSize), func(t *testing.T) {
 			runTest(ebpf.Hash, test.size, test.mapSize, t, test.opts...)
 		})
-		t.Run(fmt.Sprintf("size=%d mapSize=%d", test.size, test.mapSize), func(t *testing.T) {
-			runTest(ebpf.LRUHash, test.size, test.mapSize, t, test.opts...)
-		})
+		if !test.unsafeLRU {
+			t.Run(fmt.Sprintf("size=%d mapSize=%d", test.size, test.mapSize), func(t *testing.T) {
+				runTest(ebpf.LRUHash, test.size, test.mapSize, t, test.opts...)
+			})
+		}
 	}
 }

